packages/common/src/access/can-edit-dataset.ts

```diff
@@ -0,0 +1,13 @@
+import { IItem, IUser } from "@esri/arcgis-rest-types";
+import { hasBasePriv } from "./has-base-priv";
+import { canEditItem } from "./can-edit-item";
+
+/**
+ * Checks if user has access to edit dataset in Hub
+ * @param {IItem} model
+ * @param {IUser} user
+ * @returns {boolean}
+ */
+export function canEditDataset(model: IItem, user?: IUser): boolean {
```

Comment:
is user really optional?

Reply:
it is in the sense that, if you don't provide user you get `false`

Reply:
imo, it's better handled here as an optional arg than in the ember app. `session.currentUser` can be undefined, so by making it optional we don't have to gate these methods there

Reply:
i'd argue the client shouldn't call this method if the currentUser is not available.

Reply:
I'm nuking this in favor of calling `canEditItem` for datasets, until a different more specific check is needed